nemo/collections/asr/parts/submodules/tdt_beam_decoding.py

```diff
@@ -0,0 +1,793 @@
+                     For accurate greedy results, please use GreedyTDTInfer or GreedyBatchedTDTInfer."""
+                )
+            self.search_algorithm = self.default_beam_search
+        elif search_type == "tsd":
+            raise NotImplementedError("`tsd` (Time Synchronous Decoding) has not been implemented.")
+        elif search_type == "alsd":
+            raise NotImplementedError("`alsd` (Alignment Length Synchronous Decoding) has not been implemented.")
+        elif search_type == "nsc":
+            raise NotImplementedError("`nsc` (Constrained Beam Search) has not been implemented.")
+        elif search_type == "maes":
+            self.search_algorithm = self.modified_adaptive_expansion_search
+        else:
+            raise NotImplementedError(
+                f"The search type ({search_type}) supplied is not supported!\n" f"Please use one of : (default, maes)"
+            )
+
+        if self.search_type == 'maes':
+            self.maes_num_steps = int(maes_num_steps)
+            self.maes_prefix_alpha = int(maes_prefix_alpha)
+            self.maes_expansion_beta = int(maes_expansion_beta)
+            self.maes_expansion_gamma = float(maes_expansion_gamma)
+
+            self.max_candidates += maes_expansion_beta
+
+            if self.maes_prefix_alpha < 0:
+                raise ValueError("`maes_prefix_alpha` must be a positive integer.")
+
+            if self.vocab_size < beam_size + maes_expansion_beta:
+                raise ValueError(
+                    f"beam_size ({beam_size}) + expansion_beta ({maes_expansion_beta}) "
+                    f"should be smaller or equal to vocabulary size ({self.vocab_size})."
+                )
+
+            if self.maes_num_steps < 2:
+                raise ValueError("`maes_num_steps` must be greater than 1.")
+
+        try:
+            self.zero_duration_idx = self.durations.index(0)
+        except ValueError:
+            self.zero_duration_idx = None
+        self.min_non_zero_duration_idx = int(
+            np.argmin(np.ma.masked_where(np.array(self.durations) == 0, self.durations))
+        )
+
+        if ngram_lm_model:
+            if search_type != "maes":
+                raise ValueError("For decoding with language model `maes` decoding strategy must be chosen.")
+
+            if KENLM_AVAILABLE:
+                self.ngram_lm = kenlm.Model(ngram_lm_model)
+                self.ngram_lm_alpha = ngram_lm_alpha
+            else:
+                raise ImportError(
+                    "KenLM package (https://github.com/kpu/kenlm) is not installed. " "Use ngram_lm_model=None."
+                )
+        else:
+            self.ngram_lm = None
+
+    @typecheck()
+    def __call__(
+        self,
+        encoder_output: torch.Tensor,
+        encoded_lengths: torch.Tensor,
+        partial_hypotheses: Optional[List[Hypothesis]] = None,
+    ) -> Union[Hypothesis, NBestHypotheses]:
```

Comment:
This type annotation is incorrect according to the real return value.
Should be something like `tuple[list[Hypothesis | NBestHypotheses],]` (tuple containing one element - a list of `Union[Hypothesis, NBestHypotheses]`)